### PR TITLE
creatibutor: implement component according to latest mockups

### LINKED ISSUE #154: creators: implement component
### Mockups
We have 3 different views for the creator component:

- List of creators view
Screenshot 2021-01-15 at 14 41 03

- Add a new creator modal
Screenshot 2021-01-15 at 15 20 59

- Edit a creator modal
Screenshot 2021-01-15 at 15 21 08

### Use cases
A user should be able to:

- Add a new creator
- Remove a creator
- Edit the metadata of a specific creator
- Order the specified creators
- Add single creator
- Add multiple creators in an easy way without too many clicks
- Should be able to cancel the addition of a creator

#### List of creators modal
- [ ] Imlement a list item view to display each creator
  - [ ] Display information `name` or `family name` and `given name`, affiliation name, orcid icon if orcidis specified
  - [x] Order creators. I
      - [x] Implement a drag-n-drop functionality to rearrange creators. 
      - [ ] Ensure the order of creators specified in the UI is respected when posting to backend
  - [x] Edit creators
      - [x]  Button that opens a modal with specified creator's information prefilled.
  - [x] Button to remove creator from the specified list
- [x] "Add creator" button
  - [ ] Opens a modal with an empty form to add a new creator

#### Add a creator modal

- [x] Modal should be disposable i.e close on `dimmer` click.
- [ ] Modal header
  - [x] "Add creator" label on the right
  - [ ] "close" icon on the right
- [ ] Modal body - New creators form
  - [x] `type` field should be toggle
  - [x] `Family name` and `given name` should be show when the toggle value is `Person`. Otherwise, we display one input field for `name`
  - [x] Identifiers should be an input field that can display multiple values. Click and add tag functionaily.
  - [ ] Affiliations should be an autocomplete field from a specified vocabulary. No custom affiliation will be supported currently.
  - [ ] Role should be a dropdown field. Potentially autocomplete from vocabulary it's options if available.
- [ ]  Modal footer
  - [x] "Cancel" button on left of the side. Will discard any action and close the modal
  - [x] "Save and add another" button on the right side. The button will:
    - [ ] Validate input data for creator
    - [x] Save input data and show in the deposit form the new creator
    - [x] Clear the modal form
    - [ ] Focus on `Family name` input field, ready for next input.
    - [ ] Show a feedback message that creator was saved.
  - [ ] "Save"  button. The button will:
    - [ ]  Validate input data for creator
    - [x] Save input data and show in the deposit form the new creator
    - [x] Close modal.
 
#### Edit creator modal

Differences with modal implementation above

- [x] Modal header should display "Edit creator"
- [x] Modal form prefilled with information
- [x] Modal foot should contain only `Cancel` and `Save` buttons. Same position.
- [x] Modal should be disposable i.e close on `dimmer` click.

### PR DESCRIPTION
**Pull request by zzacharo**
closes #154 

requires https://github.com/inveniosoftware/react-invenio-forms/pull/46 to be merged first

### CODE REVIEW

File: src/lib/components/Creatibutors/CreatibutorsModal.js [hunk 1]
```diff
@@ -0,0 +1,348 @@
...
+          const affiliationsFieldPath = `${basePath}.affiliations`;
+          const roleFieldPath = `${basePath}.role`;
+          return (
+            <Modal
+              onOpen={() => this.openModal()}
+              open={this.state.open}
+              trigger={this.props.trigger}
+              onClose={this.closeModal}
+            >
+              <Modal.Header as="h6" className="license-modal-header">
```
Comment (zzacharo) [inline src/lib/components/Creatibutors/CreatibutorsModal.js:165]: review classname

File: src/lib/DepositRecordSerializer.js [hunk 2]
```diff
@@ -32,7 +32,6 @@ import {
   VocabularyField,
 } from './fields';
 
-
```
Comment (fenekku) [inline src/lib/DepositRecordSerializer.js:35]: Minor: I guess I am guilty here as I have been aiming for there to be 2 new lines after imports and before the first class declaration like in Python. Is there a style guide I could follow? (I believe I tried installing prettier at some point, but didn't manage to make it work on my computer. Is it even the reference we use?)

File: src/lib/DepositRecordSerializer.js [hunk 3]
```diff
@@ -49,14 +48,16 @@ export class DepositRecordSerializer {
       fieldpath: 'metadata.additional_titles',
       deserializedDefault: [],
     }),
-    creators: new CreatibutorsField({
+    creators: new Field({
       fieldpath: 'metadata.creators',
-      deserializedDefault: [emptyCreatibutor],
+      // deserializedDefault: [{ type: 'personal' }],
```
Comment (fenekku) [inline src/lib/DepositRecordSerializer.js:53]: remove?

File: src/lib/components/CreatibutorsField.js [hunk 4]
```diff
@@ -8,7 +8,12 @@ import _get from 'lodash/get';
 import React, { Component } from 'react';
```
Comment (fenekku) [inline src/lib/components/CreatibutorsField.js:8]: Is this file still valid? There is a file of same name in `src/lib/components/Creatibutors/CreatibutorsField.js` above. I am assuming this one can be deleted.

File: src/lib/fields/CreatibutorsField.js [hunk 5]
```diff
@@ -12,36 +12,6 @@ import _set from 'lodash/set';
 import { Field } from './Field';
```
Comment (fenekku) [inline src/lib/fields/CreatibutorsField.js:12]: This file is not used in RecordSerializer.js anymore, but only some of the code was deleted. Should it still be used? If not, we should delete this file.

File: src/lib/components/Creatibutors/CreatibutorsField.js [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+
+export class CreatibutorsField extends Component {
+  render() {
+    return (
+      <FieldArray
+        name={this.props.fieldPath}
+        test={'test'}
+        component={(formikProps) => (
+          <CreatibutorsFieldForm {...formikProps} {...this.props} />
+        )}
```
Comment (fenekku) [inline src/lib/components/Creatibutors/CreatibutorsField.js:121]: :+1: I think I prefer this technique for Formik wrapped field. It allows us to control the props passed down more effectively. It's error-prone and detrimental to understanding when the wrapped field is a method on the same class and therefore needs to pop some props to make sure they are not used in the `...uiProps` pattern we have been using.

File: src/lib/components/Creatibutors/CreatibutorsIdentifiers.js [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+// React-Invenio-Deposit is free software; you can redistribute it and/or modify it
+// under the terms of the MIT License; see LICENSE file for more details.
+
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { SelectField } from 'react-invenio-forms';
+import _unickBy from 'lodash/unionBy';
+
+export class CreatibutorsIdentifiers extends Component {
+  static propTypes = {
```
Comment (fenekku) [inline src/lib/components/Creatibutors/CreatibutorsIdentifiers.js:14]: :warning: This doesn't seem to be supported in Safari https://caniuse.com/mdn-javascript_classes_static_class_fields . Can they be assigned like we usually do?

**Review by fenekku — APPROVED** [review]
Honestly, I am "assuming" this makes sense for most of it. There is a lot of code dealing with some conversion between different representations. I would be careful there to see if we can go without it. My take is that formik should be our data store (not the components). Any interaction should just modify formik.form.values. However, it may be required - don't know if we can judge that right now.

Because this is complex I will not attempt to touch the code and let you merge / address comments as you see fit.

File: src/lib/DepositRecordSerializer.js (hunk 2 above)
Comment (zzacharo) [reply src/lib/DepositRecordSerializer.js:35]: I am just using prettier with the default configuration. Let's have a talk about it and align :)

File: src/lib/components/CreatibutorsField.js (hunk 4 above)
Comment (zzacharo) [reply src/lib/components/CreatibutorsField.js:8]: Removed!

File: src/lib/fields/CreatibutorsField.js (hunk 5 above)
Comment (zzacharo) [reply src/lib/fields/CreatibutorsField.js:12]: Removed!

File: src/lib/components/Creatibutors/CreatibutorsIdentifiers.js (hunk 7 above)
Comment (zzacharo) [reply src/lib/components/Creatibutors/CreatibutorsIdentifiers.js:14]: I tested it and it works...maybe babel or react traspiles it correctly out of the box as it is documented on their side!

Comment (fenekku) [reply src/lib/components/Creatibutors/CreatibutorsIdentifiers.js:14]: ah good!

**Pull request merged**